### PR TITLE
feat(ns-api): add ns.report latency-and-quality-report

### PR DESCRIPTION
**Pull request by andre8244**
Ref:
- https://github.com/NethServer/nethsecurity/issues/904

### CODE REVIEW

File: packages/ns-api/files/ns.report [hunk 1]
```diff
@@ -315,13 +316,82 @@ def ovpnrw_bytes_by_hour_and_user(instance, day, user):
...
+        data = response.read()
+        json_data = json.loads(data)
+
+    # find charts that start with 'fping'
+    if 'charts' in json_data:
+        matching_keys = [key for key in json_data['charts'].keys() if key.startswith('fping')]
+
+    return matching_keys
+
+
```
Comment (gsanchietti) [inline packages/ns-api/files/ns.report:331]: This function is not used

File: packages/ns-api/files/ns.report [hunk 2]
```diff
@@ -315,13 +316,82 @@ def ovpnrw_bytes_by_hour_and_user(instance, day, user):
...
+def latency_and_quality_report():
+    hosts = get_fping_hosts()
+    ret = {}
+
+    for host in hosts:
+        host_replaced = host.replace('.', '_')
+
+        # latency chart
+        latency_chart_name = f'fping.{host_replaced}_latency'
+        latency_chart_data = get_netdata_chart_data(latency_chart_name)
```
Comment (gsanchietti) [inline packages/ns-api/files/ns.report:373]: Probably a oneliner is readable enough here:
```suggestion
        latency_chart_data = get_netdata_chart_data(f'fping.{host_replaced}_latency')
```

File: packages/ns-api/files/ns.report [hunk 3]
```diff
@@ -315,13 +316,82 @@ def ovpnrw_bytes_by_hour_and_user(instance, day, user):
...
+    url = f'http://127.0.0.1:19999/api/v1/data?chart={chart_name}&after=-3600&points=180&options=abs'
+    try:
+        with urllib.request.urlopen(url) as fu:
+            data = json.loads(fu.read())
+    except:
+        return ret
+
+    for record in data["data"]:
+        # first element is the label
+        ret["labels"].append(record[0])
```
Comment (gsanchietti) [inline packages/ns-api/files/ns.report:357]: Are you sure that labels are not inside `data['labels']` field?

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: packages/ns-api/files/ns.report (hunk 3 above)
Comment (andre8244) [reply packages/ns-api/files/ns.report:357]: Yes we are referring to the chart label, i.e. the value on x axis (the timestamp); let's improve the comment for clarity

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**